### PR TITLE
fix(pii): Fix issue where `$span` would not be recognized in Advanced Data Scrubbing

### PR DESCRIPTION
**Pull request by untitaker**

### CODE REVIEW

File: relay-general/src/macros.rs [hunk 1]
```diff
@@ -98,3 +98,56 @@ macro_rules! impl_str_serde {
...
+///     Bar,
+/// }
+///
+/// derive_fromstr_and_display!(ValueType, ValueTypeError, {
+///     ValueType::Foo => "foo" | "foo2",  // fromstr will recognize foo/foo2, display will use foo.
+///     ValueType::Bar => "bar",
+/// });
+/// ```
+macro_rules! derive_fromstr_and_display {
+    ($type:ty, $error_type:tt, { $($variant:path => $($name:literal)|*),* $(,)* }) => {
```
Comment (jan-auer) [inline relay-general/src/macros.rs]: I didn't run it, but this should avoid invocations like `derive_fromstr_and_display!(A, B, ,,,)`

```suggestion
    ($type:ty, $error_type:tt, { $($variant:path => $($name:literal)|*),+ $(,)? }) => {
```

**Review by jan-auer — APPROVED** [review]
The macro will potentially obstruct missing variant errors, but is a clear improvement over what we had before.

**Comment by untitaker** [comment]
>potentially obstruct missing variant errors

It will fail compilation for sure but I have no idea if the error message is pretty

**Comment by untitaker** [comment]
ftr, as per slack suggestion I looked into strum and found it harder to navigate wrt what values get used for Display and which values are allowed. This is from the wiki for Display:

>Of the various serialize properties, the value with the longest length is chosen. If that behavior isn't desired, you should use to_string.

https://github.com/Peternator7/strum/wiki/Derive-Display

That's not what we want, so we'd need to set to_string for ValueType all the time.

**Comment by jan-auer** [comment]
What if we merge this in and look into `strum` separately? What you have now is undoubtedly less error-prone.

**Comment by untitaker** [comment]
blocked on #780

**Pull request merged**